zebra-utils/src/bin/scanning-results-reader/main.rs

```diff
@@ -0,0 +1,128 @@
+///
+/// Reads the results storage, iterates through all decrypted memos, and prints the them to standard
+/// output. Filters out some frequent and uninteresting memos typically associated with ZECPages.
+///
+/// Notes:
+///
+/// - `#[allow(clippy::print_stdout)]` is set to allow usage of `println!` for displaying the memos.
+/// - This function expects Zebra's RPC server to be available.
+///
+/// # Panics
+///
+/// When:
+///
+/// - The Sapling key from the storage is not valid.
+/// - There is no diversifiable full viewing key (dfvk) available.
+/// - The RPC response cannot be decoded from a hex string to bytes.
+/// - The transaction fetched via RPC cannot be deserialized from raw bytes.
+#[allow(clippy::print_stdout)]
+pub fn main() {
+    let network = zcash_primitives::consensus::Network::MainNetwork;
+    let storage = Storage::new(&Config::default(), zebra_network(&network), true);
+    let mut prev_memo = "".to_owned();
+
+    for (key, _) in storage.sapling_keys_last_heights().iter() {
+        let dfvk = sapling_key_to_scan_block_keys(key, zebra_network(&network))
+            .expect("Scanning key from the storage should be valid")
+            .0
+            .into_iter()
+            .next()
+            .expect("There should be exactly one dfvk");
+
+        let ufvk_with_acc_id = HashMap::from([(
+            AccountId::from(1),
+            UnifiedFullViewingKey::new(Some(dfvk), None).expect("`dfvk` should be `Some`"),
+        )]);
+
+        for (height, txids) in storage.sapling_results(key) {
+            let height = BlockHeight::from_u32(height.0);
+
+            for txid in txids.iter() {
+                let tx = Transaction::read(
+                    &hex::decode(&get_tx_via_rpc(hex::encode(txid.bytes_in_display_order())))
+                        .expect("RPC response should be decodable from hex string to bytes")[..],
+                    BranchId::for_height(&network, height),
+                )
+                .expect("TX fetched via RPC should be deserializable from raw bytes");
+
+                for output in decrypt_transaction(&network, height, &tx, &ufvk_with_acc_id) {
+                    let memo = memo_bytes_to_string(output.memo.as_array());
+
+                    if !memo.is_empty()
+                        // Filter out some uninteresting and repeating memos from ZECPages.
+                        && !memo.contains("LIKE:")
+                        && !memo.contains("VOTE:")
+                        && memo != prev_memo
+                    {
+                        println!("{memo}\n");
+                        prev_memo = memo;
+                    }
+                }
+            }
+        }
+    }
+}
+
+/// Trims trailing zeroes from a memo, and returns the memo as a string.
+fn memo_bytes_to_string(memo: &[u8; 512]) -> String {
+    match memo.iter().rposition(|&byte| byte != 0) {
+        Some(i) => std::str::from_utf8(&memo[..=i]).unwrap_or("").to_owned(),
```

Comment:
Optional: This will ignore some invalid UTF-8 sequences, which might or might not be what we want. Let's test and see?

```suggestion
        Some(i) => std::str::from_utf8_lossy(&memo[..=i]).into_owned(),
```

Reply:
Done in [c0c2101](https://github.com/ZcashFoundation/zebra/pull/8104/commits/c0c21016f13f91b63060152cdaabd982c5901354).